toolkit/hooks/src/use-progress.ts

```diff
@@ -0,0 +1,95 @@
+import React, { useState, useEffect, useRef } from "react";
+
+interface UseProgresInterface {
+    progressInterval?: number,
+    running: boolean
+    defaultProgress?: number
+    totalFileSize?: number,
+    processedFileSize?: number
+
+}
+
+const useProgress = ({
+  progressInterval = 4000,
+  running = false,
+  defaultProgress = 0,
+  totalFileSize = 0,
+  processedFileSize = 0,
+}: UseProgresInterface) => {
+  const [progress, setProgress] = useState<number>(defaultProgress);
+  const [currentFileProcessed, setCurrentFileProcessed] = useState<number>(0);
+
+  let currentProgressRef = useRef<number>(0);
+
+  const intervalRef = useRef<NodeJS.Timeout | null>(null);
+  const increment = (100 / progressInterval) * 12;
+
+  const updateProgress = () => {
+    if (totalFileSize === 0) {
+      currentProgressRef.current += increment;
+      const currentProgress = Number(currentProgressRef.current.toFixed(0));
+      setProgress(Math.min(currentProgress, 100));
+    }
+    if (totalFileSize > 0) {
+      currentProgressRef.current = currentFileProcessed;
+      setProgress(currentProgressRef.current);
+    }
+  };
+
+  const start = () => {
+    intervalRef.current = setInterval(() => {
+      updateProgress();
+    }, 10);
+  };
+
+  const pause = () => {
+    if (progress > 0 && progress < 100) {
+      const currentProgress:number = Number(currentProgressRef.current.toFixed(0));
+      setProgress(currentProgress);
+      clearInterval(intervalRef.current as NodeJS.Timeout);
```

Comment:
NodeJS is used here but not imported anywhere in this file.

Reply:
this is added in tsconfig.json file that's why no need to import it ,automatically detects and not showing any error for that.